orkaudio/filters/LiveStream/LiveStreamServerProxy.cpp

```diff
@@ -0,0 +1,207 @@
+ *
+ * http://www.orecx.com
+ *
+ * This program is free software, distributed under the terms of
+ * the GNU General Public License.
+ * Please refer to http://www.gnu.org/copyleft/gpl.html
+ *
+ */
+#ifdef WIN32
+#include <WinSock2.h>
+#include <WS2tcpip.h>
+#include <Windows.h>
+//#include "winsock2.h"
+#endif
+#include "LiveStreamServerProxy.h"
+#include "ConfigManager.h"
+#include "CapturePort.h"
+#include <set>
+
+LiveStreamServerProxy *LiveStreamServerProxy::m_singleton;
+
+LiveStreamServerProxy::LiveStreamServerProxy()
+{
+	m_runFunction = NULL;
+	m_startStreamFunction = NULL;
+	m_endStreamFunction = NULL;
+	m_getStreamFunction = NULL;
+	m_loaded = false;
+}
+
+LiveStreamServerProxy *LiveStreamServerProxy::Singleton()
+{
+	return m_singleton;
+}
+
+bool LiveStreamServerProxy::Initialize()
+{
+	m_singleton = new LiveStreamServerProxy();
+	return m_singleton->Init();
+}
+
+bool LiveStreamServerProxy::Init()
+{
+	OrkAprSubPool locPool;
+
+	// Get the desired capture plugin from the config file, or else, use the first dll encountered.
+	CStdString pluginDirectory = CONFIG.m_capturePluginPath + "/";
+	CStdString pluginPath;
+	apr_status_t ret;
+	if (!CONFIG.m_capturePlugin.IsEmpty())
+	{
+		// A specific plugin was specified in the config file
+		pluginPath = pluginDirectory + "orkaudio/plugins/livestream.so";
+	}
+	else
+	{
+		// No plugin specified, find the first one in the plugin directory
+		apr_dir_t *dir;
+		ret = apr_dir_open(&dir, (PCSTR)pluginDirectory, AprLp);
+		if (ret != APR_SUCCESS)
+		{
+			LOG4CXX_ERROR(LOG.rootLog, CStdString("Capture plugin directory could not be found:" + pluginDirectory));
+		}
+		else
+		{
+			apr_finfo_t finfo;
+			apr_int32_t wanted = APR_FINFO_NAME | APR_FINFO_SIZE;
+			while ((ret = apr_dir_read(&finfo, wanted, dir)) == APR_SUCCESS)
+			{
+				CStdString fileName;
+				fileName.Format("%s", finfo.name);
+				if (fileName.find(".dll") != std::string::npos)
+				{
+					pluginPath = pluginDirectory + finfo.name;
+					break;
+				}
+			}
+			apr_dir_close(dir);
+		}
+	}
+	if (!pluginPath.IsEmpty())
+	{
+		apr_status_t ret;
+		char errstr[256];
+		// dso handle needs to persist so it must me allocated using a pool that
+		// will also persist.
+		ret = apr_dso_load(&m_dsoHandle, (PCSTR)pluginPath, OrkAprSingleton::GetInstance()->GetAprMp());
+		if (ret != APR_SUCCESS)
+		{
+			apr_strerror(ret, errstr, sizeof(errstr));
+			apr_dso_error(m_dsoHandle, errstr, sizeof(errstr));
+#ifndef WIN32
+			LOG4CXX_ERROR(LOG.rootLog, CStdString("Failed to load the following plugin: ") + pluginPath + " " + errstr + CStdString(", could be missing dependency, Try running in cmd box (orkaudio debug)"));
+#else
+			LOG4CXX_ERROR(LOG.rootLog, CStdString("Failed to load the following plugin: ") + pluginPath + CStdString(", could be missing dependency. Try running ldd on it."));
+#endif
+		}
+		else
+		{
+			// Ok, the dll has been successfully loaded
+			LOG4CXX_INFO(LOG.rootLog, CStdString("Loaded plugin: ") + pluginPath);
```

Comment:
Why does LiveStream code, load it's on `.so` dynamically? This could be simplified to normal dependency IMO.

Reply:
these changes are non needed anymore